crates/uv-resolver/src/python_requirement.rs

```diff
@@ -59,10 +59,32 @@ impl PythonRequirement {
         self.target.as_ref()
     }
 
-    /// Return the target version of Python as a "requires python" type,
-    /// if available.
-    pub(crate) fn requires_python(&self) -> Option<&RequiresPython> {
-        self.target().and_then(|target| target.as_requires_python())
+    /// Return a [`MarkerTree`] representing the Python requirement.
+    ///
+    /// See: [`RequiresPython::to_marker_tree`]
+    pub fn to_marker_tree(&self) -> MarkerTree {
+        let version = match &self.target {
+            None => self.installed.version.clone(),
+            Some(PythonTarget::Version(version)) => version.version.clone(),
+            Some(PythonTarget::RequiresPython(requires_python)) => {
+                return requires_python.to_marker_tree()
+            }
+        };
+
+        let version_major_minor_only = Version::new(version.release().iter().take(2));
+        let expr_python_version = MarkerExpression::Version {
+            key: MarkerValueVersion::PythonVersion,
+            specifier: VersionSpecifier::from_version(Operator::Equal, version_major_minor_only)
+                .unwrap(),
+        };
+        let expr_python_full_version = MarkerExpression::Version {
+            key: MarkerValueVersion::PythonFullVersion,
+            specifier: VersionSpecifier::from_version(Operator::Equal, version).unwrap(),
```

Comment:
I'm a bit unsure of this. It still locks against the target Python version rather than "the target version at minimum." It's easy to change but not sure what the right semantics are.

Reply:
My feeling is:
* `requires-python` if the input is a `pyproject.toml` (the conjunction of all `requires-python` values if there are multiple `pyproject.toml`).
* An explicit `--requires-python` argument.
* `>={{current python}}` and inform the user that the should consider passing `--requires-python` if they want an interpreter independent resolution.